components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTx.java

```diff
@@ -37,14 +38,25 @@
     extends Transaction
 {
   /**
-   * Provides the underlying graph transaction.
+   * Provides the underlying document transaction.
    *
-   * Note: The caller may use this to directly query or manipulate the OrientDB graph, if needed, but should not
+   * Note: The caller may use this to directly query or manipulate the Orient document DB, if needed, but should not
    * directly commit, rollback, or close the underlying transaction.
    */
   ODatabaseDocumentTx getDb();
 
   /**
+   * A lower level but still safe access to SQL SELECT queries, that allows caller to "browse" potentially
+   * huge result sets without worry about paging or OOMs. This method will NOT see uncommited changes performed
+   * in this current TX, if any.
+   *
+   * @param selectSql The SELECT SQL in full, optionally with named parameters.
+   * @param params The map holding named parameters of SELECT SQL.
+   * @return Iterable of SELECTed {@link ODocument}s.
```

Comment:
Should also warn that a runtime exception will be thrown from the iteration if the timeout is breached.